shell/platform/darwin/ios/ios_render_target_gl.mm

```diff
@@ -12,21 +12,25 @@
 
 namespace flutter {
 
-IOSGLRenderTarget::IOSGLRenderTarget(fml::scoped_nsobject<CAEAGLLayer> layer,
-                                     EAGLContext* context,
-                                     EAGLContext* resource_context)
-    : layer_(std::move(layer)),
-      context_([context retain]),
-      resource_context_([resource_context retain]),
-      framebuffer_(GL_NONE),
-      colorbuffer_(GL_NONE),
-      storage_size_width_(0),
-      storage_size_height_(0),
-      valid_(false) {
+static fml::scoped_nsobject<CAEAGLLayer> CastEAGLLayer(fml::scoped_nsobject<CALayer> layer) {
+  FML_DCHECK([layer isKindOfClass:[CAEAGLLayer class]]);
+  return fml::scoped_nsobject<CAEAGLLayer>{reinterpret_cast<CAEAGLLayer*>([layer.get() retain])};
+}
+
+IOSRenderTargetGL::IOSRenderTargetGL(fml::scoped_nsobject<CALayer> layer,
+                                     fml::scoped_nsobject<EAGLContext> context,
+                                     fml::scoped_nsobject<EAGLContext> resource_context)
+    : layer_(CastEAGLLayer(std::move(layer))),
```

Comment:
It probably makes more sense to make this method take in a EAGLLayer and push the cast higher in the callstack.